packages/find-root/src/index.ts

```diff
@@ -2,6 +2,8 @@ import findUp, { sync as findUpSync } from "find-up";
 import path from "path";
 import fs from "fs-extra";
 
+import { Tool, ToolType, NoneTool, defaultOrder, supportedTools, MonorepoRoot } from "@manypkg/core";
```

Comment:
I would probably try to keep the `@manypkg/core` (`/tools`?) as lean as possible. For that reason I would make the `defaultOrder` responsibility of the consumer - and in this case, that would be here, in the `@manypkg/find-root`.

Reply:
I can definitely be convinced of this.

Here's my concern -- you've addressed a few corner cases in the existing tools by using "carve outs" in the logic, and I wonder if that is really appropriate. (As an example: LernaTool skips itself if lerna.json specifies useWorkspaces, which lets the logic fall back to YarnTool or PnpmTool.)

It seems like what's more appropriate is to recognize that as the number of Tools grows, you'll end up with Tools that are overlapping or are strict subsets/supersets of each other; the "default order" is an important part of the list of tools, because it is the order that manypkg guarantees will pick the most specific Tool for your repo, falling back to less specific tools.

If we rely on the consumer of the tools to provide a default order, they need to revisit the list every time they update the dependency.

Reply:
We could do this:
1. move `defaultOrder` to `find-root`
2. require the user to opt into keyed tools with smth like `{ rush: true }`. 

This way we would still be in charge of the order while also allowing the user to specify what they allow to be detected.